hw2/Views/LeagueView.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  LeagueView.swift
+//  hw2
+//
+//  Created by Tina Jureško on 14.03.2025..
+//
+
+import Foundation
+import UIKit
+import SofaAcademic
+import SnapKit
+
+class LeagueView: BaseView {
+    
+    private let logoImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: "LaLiga")
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+    
+    private let countryLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    private let arrowImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: "Vector")
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+    
+    private let nameLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+        
+    override func addViews() {
+        super.addViews()
+        addSubview(logoImageView)
+        addSubview(countryLabel)
+        addSubview(arrowImageView)
+        addSubview(nameLabel)
+    }
+
+    override func styleViews() {
+        logoImageView.contentMode = .scaleAspectFit
+        arrowImageView.contentMode = .scaleAspectFit
+        
+        countryLabel.font = .regularBold14()
+        countryLabel.textColor = AppStyles.Colors.primary
+        countryLabel.numberOfLines = 0
+        countryLabel.lineBreakMode = .byWordWrapping
+        
+        nameLabel.font = .regularBold14()
+        nameLabel.textColor = AppStyles.Colors.secondary
+        nameLabel.numberOfLines = 0
+        nameLabel.lineBreakMode = .byWordWrapping
```

Comment:
Trebalo bi na svaku liniju staviti lineHeight = 16 ili vec kako definirano u dizajnu. 

Mozes da ne dupliciras uvijek tu liniju napraviti i ovako nesto.

public extension UILabel {

class Style {

    public let font: UIFont
    public let lineHeight: CGFloat?

    public init(font: UIFont, lineHeight: CGFloat? = nil) {
        self.font = font
        self.lineHeight = lineHeight
    }
}

func setStyle(_ style: Style?) {
    font = style?.font
    lineHeight = style?.lineHeight
}
}

I onda u style view na svaki lable postavljas style umjesto fonta, a style unutra sebe ima regular14 npr.
label.style = .regular14

Ako ne to, onda je preglednije samo na svaki label postaviti lineHeight = 16. Zelimo constrainti budu sto jednostavniji i citljiviji